app/code/Magento/Captcha/Test/Unit/Observer/CheckUserLoginBackendObserverTest.php

```diff
@@ -0,0 +1,178 @@
+        $captchaValue = 'captcha-value';
+
+        /** @var Observer|MockObject $observerMock */
+        $observerMock = $this->createPartialMock(Observer::class, ['getEvent']);
+        $eventMock = $this->createPartialMock(Event::class, ['getUsername']);
+        $captcha = $this->createMock(DefaultModel::class);
+
+        $eventMock->expects($this->any())
+            ->method('getUsername')
+            ->willReturn('admin');
+        $observerMock->expects($this->any())
+            ->method('getEvent')
+            ->willReturn($eventMock);
+        $captcha->expects($this->once())->method('isRequired')
+            ->with($login)
+            ->willReturn($isRequired);
+        $captcha->expects($this->exactly($invokedTimes))
+            ->method('isCorrect')
+            ->with($captchaValue)
+            ->willReturn($isCorrect);
+        $this->helperMock->expects($this->once())
+            ->method('getCaptcha')
+            ->with($formId)
+            ->willReturn($captcha);
+
+        $this->captchaStringResolverMock->expects($this->exactly($invokedTimes))
+            ->method('resolve')
+            ->with($this->requestMock, $formId)
+            ->willReturn($captchaValue);
+
+        $this->messageManagerMock->expects($this->exactly(0))
+            ->method('addError')
+            ->with(__('Incorrect CAPTCHA'));
+
+        $this->observer->execute($observerMock);
+    }
+
+    /**
+     * @return array
+     */
+    public function captchaCorrectnessCheckDataProvider(): array
+    {
+        return [
+            [true, true, 1],
+            [false, true, 0]
+        ];
+    }
+
+
+    /**
+     * Test check user login in backend with wrong captcha
+     *
+     * @return void
+     * @throws AuthenticationException
+     */
+    public function testCheckOnBackendLoginWithWrongCaptcha(): void
+    {
+        $formId = 'backend_login';
+        $login = 'admin';
+        $captchaValue = 'captcha-value';
+
+        /** @var Observer|MockObject $observerMock */
+        $observerMock = $this->createPartialMock(Observer::class, ['getEvent']);
+        $eventMock = $this->createPartialMock(Event::class, ['getUsername']);
+        $captcha = $this->createMock(DefaultModel::class);
+
+        $eventMock->expects($this->any())
+            ->method('getUsername')
+            ->willReturn('admin');
+        $observerMock->expects($this->any())
+            ->method('getEvent')
+            ->willReturn($eventMock);
+        $captcha->expects($this->once())->method('isRequired')
+            ->with($login)
+            ->willReturn(true);
+        $captcha->expects($this->exactly(1))
+            ->method('isCorrect')
+            ->with($captchaValue)
+            ->willReturn(false);
+        $this->helperMock->expects($this->once())
+            ->method('getCaptcha')
+            ->with($formId)
+            ->willReturn($captcha);
+
+        $this->captchaStringResolverMock->expects($this->exactly(1))
```

Comment:
Too strict, irrelevant assertion.

Reply:
Hi @orlangur thank you for all the suggestions. By 'Too strict', do you mean that we should try to avoid the `expects` strictness in some cases?